### PR TITLE
[16.0][IMP] account_statement_import_sheet_file: auto transaction ID from row hash

### PR DESCRIPTION
**Pull request by siddolo**
## Description

This PR adds an option to automatically generate the transaction ID during sheet import.
When enabled, the ID is computed by hashing the row values, preventing duplicates when no explicit ID column is available.

## Changes

1) Added Automatic operation ID option in the mapping; Hide transaction_id_column when the option is enabled.
2) Generate the ID via row-value hash.
3) Updated strings and translations.

### CODE REVIEW

**Comment by siddolo** [comment]
Could you please change the title to:
`[16.0][IMP] account_statement_import_sheet_file: auto transaction ID from row hash`

Thanks

File: account_statement_import_sheet_file/views/account_statement_import_sheet_mapping.xml [hunk 1]
```diff
@@ -121,7 +122,10 @@
                                     'invisible': [('amount_type', '!=', 'absolute_value')],
                                     }"
                             />
-                            <field name="transaction_id_column" />
+                            <field
+                                name="transaction_id_column"
+                                attrs="{'invisible': [('transaction_id_auto', '=', True)]}"
```
Comment (alexey-pelykh) [inline account_statement_import_sheet_file/views/account_statement_import_sheet_mapping.xml:127]: On 18.0+, use `invisible="transaction_id_auto"` directly instead of the `attrs` dict. The `attrs` syntax is deprecated.

**Review by alexey-pelykh — COMMENTED** [review]
Thanks! The auto-ID hashing approach is clean — SHA-256 with proper value normalization handles the edge cases well.

Two things:
- Missing tests for the `transaction_id_auto` feature. At minimum: verify the hash is deterministic for the same row, and changes when row data changes.
- The view uses old-style `attrs` dict for the `invisible` condition — on 18.0, this should be `invisible="transaction_id_auto"` directly on the field element.

**Comment by mt-software-de** [comment]
@siddolo i did something similar for v18 but implemented in the base.
I think this would fix your need as well. I can create a backport. When we have a common ground. 
https://github.com/OCA/bank-statement-import/pull/909

File: account_statement_import_sheet_file/views/account_statement_import_sheet_mapping.xml (hunk 1 above)
Comment (siddolo) [reply account_statement_import_sheet_file/views/account_statement_import_sheet_mapping.xml:127]: Thanks. This PR targets 16.0, where the `attrs` dict is still the supported syntax. The direct `invisible="transaction_id_auto"` expression applies to newer Odoo series, so I kept the 16.0 form here.

**Comment by siddolo** [comment]
> @siddolo i did something similar for v18 but implemented in the base. I think this would fix your need as well. I can create a backport. When we have a common ground. #909

I think it's important to be able to configure this feature in the view.
In addition, I think it's more robust to hash all the fields.

**Comment by siddolo** [comment]
Thanks for the review.

I rebased the PR on the latest 16.0 branch and added tests covering:
- deterministic hashes for identical row values;
- different hashes when row data changes;
- automatic transaction IDs through the real sheet parser.

Regarding the view syntax: this PR targets Odoo 16.0, where `attrs` is still the correct syntax, so I kept it unchanged.

Pre-commit passes. The current Runboat build is still pending; the preceding rebased build passed. I also checked the authenticated CI logs: both Odoo and OCB stop before reaching this module's tests while loading the unrelated `account_statement_import_online_ofx` module, with `ImportError: cannot import name 'ofxhome' from 'ofxtools'`. PR #986 fails with the same shared dependency/base issue.

@alexey-pelykh, could you please review again and remove the `stale` label if the automation does not clear it?
